lib/enumerables.rb

```diff
@@ -0,0 +1,177 @@
+module Enumerable
+  # my_each
+  def my_each(&block)
+    arr = self
+    arr = arr.to_a if self.class == Range || arr.is_a?(Hash)
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i][0], arr[i][1]
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count]
+        count += 1
+      end
+      self
+    else
+      "\#<Enumerator: #{self}:my_each>"
+    end
+  end
+
+  # my_each_with_index
+  def my_each_with_index(&block)
+    arr = self
+    arr = arr.to_a if self.class == Range || arr.is_a?(Hash)
+    count = 0
+    if arr.is_a?(Hash)
+      while count < arr.length
+        yield arr[i], count
+        count += 1
+      end
+      self
+    elsif block
+      while count < arr.length
+        yield arr[count], count
+        count += 1
+      end
+      self
+    else
+      "\#<Enumerator: #{self}:my_each_with_index>"
+    end
+  end
+
+  # my_select
+  def my_select(&block)
+    return "\#<Enumerator: #{self}:my_select>" unless block
+
+    temp = []
+    to_a.my_each { |num| temp.push(num) if yield(num) }
+    temp
+  end
+
+  # my_all
+  def my_all?(paramet = nil)
+    arr = self
+    arr = arr.to_a
+    return true if arr.empty?
+
+    if block_given?
+      arr.my_each do |n|
+        ans = yield n
+        return false if ans == false
+      end
+    elsif paramet.nil?
+      arr.my_each { |n| return false if n.nil? || !n }
+    elsif paramet.is_a?(Regexp)
+      arr.my_each { |x| return false unless x.include?(paramet) }
```

Comment:
In this line you're relying on your `#my_each` method to process a `Regexp`, but the tests show the following problem:

```
 4) enumerables #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
     Failure/Error: arr.my_each { |x| return false unless x.include?(paramet) }
     
     TypeError:
       no implicit conversion of Regexp into String
     # ./lib/enumerables.rb:69:in `include?'
     # ./lib/enumerables.rb:69:in `block in my_all?'
     # ./lib/enumerables.rb:15:in `my_each'
     # ./lib/enumerables.rb:69:in `my_all?'
     # ./spec/enumerables_spec.rb:187:in `block (4 levels) in <top (required)>'

  5) enumerables #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
     Failure/Error: arr.my_each { |x| return false unless x.include?(paramet) }
     
     TypeError:
       no implicit conversion of Regexp into String
     # ./lib/enumerables.rb:69:in `include?'
     # ./lib/enumerables.rb:69:in `block in my_all?'
     # ./lib/enumerables.rb:15:in `my_each'
     # ./lib/enumerables.rb:69:in `my_all?'
     # ./spec/enumerables_spec.rb:191:in `block (4 levels) in <top (required)>'
```

There are two possible solutions here:

1. Prepare `#my_each` to handle regular expressions.
2. Write a new method that handles regular expressions and all other possible cases of `#my_all`